extensions/amp-analytics/0.1/test/test-events.js

```diff
@@ -2052,6 +2052,111 @@ describes.realWin('Events', {amp: 1}, (env) => {
             totalVisibleTime: 15,
           });
         });
+
+        it('should fire event for all elements of the selector', async () => {
+          target.classList.add('sameClass');
+          target2.classList.add('sameClass');
+          config['visibilitySpec'] = {selector: ['.sameClass']};
+          iniLoadTrackerMock
+            .expects('getElementSignal')
+            .withExactArgs('ini-load', target)
+            .returns(readyPromise)
+            .once();
+          iniLoadTrackerMock
+            .expects('getElementSignal')
+            .withExactArgs('ini-load', target2)
+            .returns(readyPromise)
+            .once();
+          visibilityManagerMock
+            .expects('listenElement')
+            .withExactArgs(
+              target,
+              config.visibilitySpec,
+              readyPromise,
+              /* createReportReadyPromiseFunc */ null,
+              saveCallback
+            )
+            .returns(unlisten)
+            .once();
+          visibilityManagerMock
+            .expects('listenElement')
+            .withExactArgs(
+              target2,
+              config.visibilitySpec,
+              readyPromise,
+              /* createReportReadyPromiseFunc */ null,
+              saveCallback2
+            )
+            .returns(unlisten2)
+            .once();
+          // Dispose function
+          res = tracker.add(analyticsElement, 'visible', config, eventResolver);
+          const unlistenReady = getAmpElementSpy.returnValues[0];
+          const unlistenReady2 = getAmpElementSpy.returnValues[1];
+          // #getAmpElement Promise
+          await unlistenReady;
+          await unlistenReady2;
+          // #assertMeasurable_ Promise
+          await macroTask();
+          await macroTask();
+          saveCallback.callback({totalVisibleTime: 10});
+          saveCallback2.callback({totalVisibleTime: 15});
+
+          // Testing that visibilty manager mock sends state to onEvent_
+          expect(eventsSpy.getCall(0).args[0]).to.equal('visible');
+          expect(eventsSpy.getCall(0).args[1]).to.equal(eventResolver);
+          expect(eventsSpy.getCall(0).args[2]).to.equal(target);
+          expect(eventsSpy.getCall(0).args[3]).to.deep.equal({
+            totalVisibleTime: 10,
+          });
+          expect(eventsSpy.getCall(1).args[0]).to.equal('visible');
+          expect(eventsSpy.getCall(1).args[1]).to.equal(eventResolver);
+          expect(eventsSpy.getCall(1).args[2]).to.equal(target2);
+          expect(eventsSpy.getCall(1).args[3]).to.deep.equal({
+            totalVisibleTime: 15,
+          });
+        });
+
+        it('should only fire one event for each element', async () => {
```

Comment:
same here, as long as we can verify the element list only contains one element we should be good.